src/store/clusterDetails.ts

```diff
@@ -0,0 +1,26 @@
+import {StateCreator} from 'zustand';
+
+import {ApiDetails} from '@services/apiEndpoint';
+
+import {connectStore, createStoreFactory} from '@store/utils';
+
+export interface ClusterDetailsSlice extends ApiDetails {
+  reset: () => void;
+  setClusterDetails: (state: ApiDetails) => void;
+}
+const initialState = {commit: '', context: '', helmchartVersion: '', namespace: '', version: '', url: ''};
```

Comment:
Why is this extracted to separate variable? It's used only once, in the actual initial state creation, so probably better to just keep it in place.

Reply:
It used twice, initially and in reset